src/test/java/eu/dissco/backend/repository/ElasticSearchRepositoryIT.java

```diff
@@ -91,83 +85,95 @@ static void initContainer() {
     client = new ElasticsearchClient(transport);
   }
 
-  @BeforeEach
-  void initRepository() {
-    repository = new ElasticSearchRepository(client, formatter);
+  @AfterAll
+  public static void closeResources() throws Exception {
+    restClient.close();
   }
 
-  @AfterEach
-  void clearIndex() throws IOException {
-    client.indices().delete(b -> b.index(INDEX));
+  private static Stream<Arguments> provideKeyValue() {
+    return Stream.of(
+        Arguments.of("digitalSpecimen.ods:physicalSpecimenId.keyword", "global_id_45634"),
+        Arguments.of("q", PREFIX + "/0")
+    );
   }
 
-  @AfterAll
-  public static void closeResources() throws Exception {
-    restClient.close();
+  @BeforeEach
+  void initRepository() {
+    repository = new ElasticSearchRepository(client);
   }
 
-  @Test
-  void testSearch() throws IOException {
+  @ParameterizedTest
+  @MethodSource("provideKeyValue")
+  void testSearch(String field, String value) throws IOException {
     // Given
-    List<DigitalSpecimenTestRecord> specimenTestRecords = new ArrayList<>();
+    List<DigitalSpecimen> specimenTestRecords = new ArrayList<>();
     String targetId = PREFIX + "/0";
-    var targetSpecimen = givenDigitalSpecimen(targetId);
-
-    for (int i = 0; i < 10; i++) {
+    var physicalId = "global_id_45634";
+    var targetSpecimen = givenDigitalSpecimen(targetId, physicalId);
+    specimenTestRecords.add(targetSpecimen);
+    for (int i = 1; i < 10; i++) {
       var specimen = givenDigitalSpecimen(PREFIX + "/" + i);
-      specimenTestRecords.add(givenDigitalSpecimenTestRecord(specimen));
+      specimenTestRecords.add(specimen);
     }
-    postDigitalSpecimens(specimenTestRecords);
+    postDigitalSpecimens(parseToElasticFormat(specimenTestRecords));
 
     // When
-    var responseReceived = repository.search(targetId, 1, 1);
+    var responseReceived = repository.search(Map.of(field, List.of(value)), 1, 1);
 
     // Then
     assertThat(responseReceived).contains(targetSpecimen);
   }
 
+  @AfterEach
```

Comment:
Should move this to the top with the rest of the setup/destroy functions

Reply:
Yes agree, will change